src/utils/resources.ts

```diff
@@ -35,3 +40,28 @@ export const referenceForObj = (obj: K8sResourceCommon) => {
   const { group, version, kind } = groupVersionKindForObj(obj);
   return referenceFor(group, version, kind);
 };
+
+/**
+ * Resolve GVK based on the provided list (or fallback if matching provider is missing).
+ */
+export const resolveProviderRef = (
+  { name, namespace }: ObjectReference,
+  providers: ProviderResource[],
+): ProviderRef => {
+  // fallback is required if provider reference inside plan is obsolete/out-of-synch
+  const fallbackProvider: Partial<ProviderResource> = {
+    apiVersion: CLUSTER_API_VERSION,
+    kind: 'Provider',
+  };
+
+  const provider = providers.find(
+    (p) => p.metadata?.namespace === namespace && p.metadata?.name === name,
+  );
+
+  return {
+    resolved: !!provider,
+    name,
```

Comment:
we now support multiple namespaces, it may be nice to return the namespace too ?

Reply:
note that `name` and `namespace` are input for resolving gvk. We have it already in the entity.